src/main/java/org/opensearch/jobscheduler/transport/ExtensionJobActionRequest.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.opensearch.jobscheduler.transport;
+
+import java.io.IOException;
+
+import org.opensearch.common.bytes.BytesReference;
+import org.opensearch.common.io.stream.BytesStreamOutput;
+import org.opensearch.common.io.stream.Writeable;
+import org.opensearch.extensions.action.ExtensionActionRequest;
+
+/**
+ * Request to extensions to invoke a job action, converts request params to a byte array
+ *
+ */
+public class ExtensionJobActionRequest<T extends Writeable> extends ExtensionActionRequest {
+
+    /**
+     * Instantiates a new ExtensionJobActionRequest
+     *
+     * @param extensionActionName the extension action to invoke
+     * @param actionParams the request object holding the action parameters
+     * @throws IOException if serialization fails
+     */
+    public ExtensionJobActionRequest(String extensionActionName, T actionParams) throws IOException {
+        super(extensionActionName, convertParamsToBytes(actionParams));
+    }
+
+    /**
+     * Takes in an object of type T that extends {@link Writeable} and converts the request to a byte array
+     *
+     * @param <T> a class that extends writeable
+     * @param actionParams the action parameters to be serialized
+     * @throws IOException if serialization fails
+     * @return the byte array of the parameters
+     */
+    public static <T extends Writeable> byte[] convertParamsToBytes(T actionParams) throws IOException {
```

Comment:
No issues with this method to produce a byte stream, but since this is effectively the same as creating a `Writeable` object, why not just pass the `actionParams` directly as a writeable object?   The only benefit I see to this is that it allows for a more flexible signature (a byte array) but that loses the type safety of the individual elements of it, and introduces complications handling the different array lengths when it's deserialized.